ENCODE_release.py

```diff
@@ -348,6 +359,61 @@ def has_inactive_pipeline(self, file_object):
                 return [p['@id'] for p in pipelines]
         return False
 
+    def _grab_bio_id(self, field, data):
+        """
+        Return biosample_term_id of embedded object.
+        """
+        return encodedcc.get_ENCODE(data.get(field),
+                                    self.connection).get('biosample_term_id')
+
+    def _get_associated_term_id(self, name, data):
+        """
+        Find biosample_term_id associated with particular object.
+        """
+        if name not in self.hela_associated_objects:
+            raise ValueError('Can only get biosample_term_id of known '
+                             'objects:', self.hela_associated_objects)
+        if name == 'File':
+            # Get biosample_term_id in file.dataset.
+            biosample_term_id = self._grab_bio_id('dataset', data)
+        elif name == 'Replicate':
+            # Get biosample_term_id in replicate.experiment.
+            biosample_term_id = self._grab_bio_id('experiment', data)
+        elif name == 'Library':
+            # Get biosample_term_id in library.biosample.
+            biosample_term_id = self._grab_bio_id('biosample', data)
+        else:
+            # For experiments and biosamples.
+            biosample_term_id = data.get('biosample_term_id')
+        return biosample_term_id
+
+    def associated_with_hela_data(self, name, data):
+        """
+        Block HeLa data if found.
+        """
+        # Define objects associated with HeLa data.
+        self.hela_associated_objects = [
+            'Experiment',
+            'Biosample',
+            'File',
+            'Library',
+            'Replicate'
+        ]
+        if name in self.hela_associated_objects:
```

Comment:
This is confusing to me.  

associated_with_hela_data only returns true if HELA is false and biosample term_id is 'EFO:0002791'.  Seems like we shouldn't even call associated_with_hela_data if self.HELA = true.  The only thing we would miss out on is the value error in _get_associated_term_id, but that will never actually trigger.

If the code functions as it should then I think we can these functions in a simpler fashion.  Let discuss tomorrow, I don't know the data so maybe I'm missing something.